src/modules/core/blob/blob-client.service.ts

```diff
@@ -0,0 +1,47 @@
+import { Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import * as mime from 'mime';
+import { IBlobClientRepository } from '@core/blob-client.interface';
+import { FileTypeInvalidError } from '../errors';
+import { UploadImageTypeEnum } from './enum';
+
+const ALLOWED_MIME_TYPES = ['image/jpeg', 'image/png'];
+
+@Injectable()
+export class BlobClientService {
+    constructor(private readonly blobClient: IBlobClientRepository, private readonly configService: ConfigService) {}
+
+    async upload(file: any, id: string, type: UploadImageTypeEnum) {
+        const directory = this.getDirectory(type);
+
+        if (!ALLOWED_MIME_TYPES.includes(file.mimetype)) {
+            throw new FileTypeInvalidError(file.mimetype, ALLOWED_MIME_TYPES);
+        }
+
+        const extension = mime.getExtension(file.mimetype);
+        // TODO: Should all these params still be passed in like the previous implementation?
+        const uploadParams = {
+            Container: this.configService.get('AZURE_CONTAINER'),
+            Key: directory + id + '.' + extension,
+            ContentType: file.mimetype,
+            Body: file.buffer,
+            ACL: 'public-read',
+            ETag: id,
+        };
```

Comment:
Most likely not, see https://learn.microsoft.com/en-us/javascript/api/@azure/storage-blob/blockblobclient?view=azure-node-latest#@azure-storage-blob-blockblobclient-upload